### ISSUE TITLE
Add ability to trigger PR test with specific tags from comment

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#1694]
I am not sure how real this idea is, but it would be nice to have possibility to trigger centos-ci tests on pull request with specific tags, all defined from a comment on the PR. It would be helpful for testing the additions and updates on feature files and test runner code, but more importantly it would help us on actual code. For example we could have just `basic.feature` as default test on PR and if anybody from reviewers would like to add any specific test, she could then define it from comment.

I am writing a change around profiles and config, reviewer could then comment, for example something like this:

```
please test tags=cmd-config,profile
```

to secure the coverage with more focused testing.

**Comment by hferentschik** [comment]
I like the idea a lot. I have been thinking about this before myself. This would improve indeed some of our workflows. 

The rub is that it won't be so easy to integrate. ATM we are using the Jenkins pull request builder. Even though it injects all sorts of environment variables, I am not sure whether we can get hold of every single commit. If so, we could handle things in the CI script for sure. See also https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin

I guess a first step would be to determine what type of information we get get hold of via the plugin. If we can get all comment data, we are home free. If not, things get more complicated.

**Comment by budhrg** [comment]
+1 for this. I also had similar ideas if we somehow pull this off, then I guess we can run think of running specific tags on default for PR and run the build using this approach for PR specific to component changes. like `proxy`. 

This will helping in reducing the CI time #1396 

WDYT? @minishift/minishift-dev

**Comment by budhrg** [comment]
@hferentschik , I don't think it is related to component/integration-test or test. It is mostly builds changes.

**Comment by gbraad** [comment]
I do not think that targeting a change to a specific set of tests is the appropriate way to reduce time in CI. All changes should undergo the same tests, regardless of where the changes have been intended. The purpose of an integration test is to ensure no regressions occur due to any change made.

It is rather helpful to retarget a certain test to happen again on a subset... as they might have failed before, but unsure about the state. Note: a re-push of a change will retrigger the build anyways for the full set...

**Comment by agajdosi** [comment]
@gbraad I agree with you - we need clearly defined set of acceptance tests, if PR does not pass them, then we can't merge. After this our perspective probably differs... What I meant (and maybe was unclear) is that reviewers could have possibility to add more features to this default set of features. And we are already in stage in which we can't execute all of the features we have.

For example, the newly added addons-xpaas.feature was not added to the set of tests on PRs. My thinking here is: if we see a PR which modifies the code around xpaas addon, it would be useful for us to have the ability of adding `@addon-xpaas` to the set of default features. It can retrigger the whole test run and on all following updates of the branch it will start the tests with this tag appended.

As I said, I am not sure if this concept was understandable in my first comment and thus I am curious what is your opinion on above paragraphs.

**Comment by LalatenduMohanty** [comment]
@agajdosi sounds fine to me. As long as we are running default set of tests of integration tests for each PR and then tags able to help us to run some extra tests we are good.

**Comment by budhrg** [comment]
Right now, its priority is quite low since this approach is not reducing any CI build time.

This will come handy when split up our integration tests into basic/smoke tests which contains most of integration tests and other specific tests.

Also, I believe the implementation of this approach will take more time.
